src/clr-angular/data/datagrid/providers/column-reorder.service.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Injectable, ViewContainerRef, ViewRef } from '@angular/core';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import { ClrDragEvent } from '../../../utils/drag-and-drop/drag-event';
+import { DatagridColumnChanges } from '../enums/column-changes.enum';
+import { ColumnsService } from './columns.service';
+import { ReorderRequest } from '../interfaces/reorder-request.interface';
+
+let nbColumnsGroup = 0;
+
+export type ReorderAnimRequest = ReorderRequest & ClrDragEvent<ViewRef>;
+
+@Injectable()
+export class ColumnReorderService {
+  // the common group id that will be shared across
+  // Datagrids all reorder draggable and droppables
+  private _columnsGroupId: string;
+
+  constructor(private columnsService: ColumnsService) {
+    this._columnsGroupId = 'dg-column-group-' + nbColumnsGroup++;
```

Comment:
Can't we store this on the ColumnsService instead of here? It might be useful in other places to have knowledge of a unique ID for a datagrid.

Reply:
I think it's better to keep this id here. There might be other Datagrid components that might make use of DnD in the future (there is actually already one which column separators). So I would like to keep this id here as it's specifically used to group reorderable parts of the datagrid columns.